From e0f002589f29d8f63dae754c7892d1c2db0de9a8 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Thu, 25 Jul 2019 11:47:47 +0530
Subject: [PATCH 0239/1921] octeontx2-af: Do not error out for debugfs partial
 reads

Debugfs file reads should return error only when copy
to user fails otherwise zero should be returned.

Change-Id: Ie80c22f16ba69bba9014a512ac9d1a8f64d26794
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/13207
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c
index e48530392d90..76714c25e1e5 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c
@@ -154,9 +154,7 @@ static ssize_t rvu_dbg_rsrc_attach_status(struct file *filp,
 
 	/* don't allow partial reads */
 	if (*ppos != 0)
-		return -EINVAL;
-	if (count < buf_size)
-		return -ENOSPC;
+		return 0;
 
 	buf = kzalloc(buf_size, GFP_KERNEL);
 	if (!buf)
-- 
2.31.1

